### ISSUE TITLE
singular4.0.3p3 compilation failure with gcc 5.4.0 in factory/cfGcdAlgExt.cc

### ISSUE THREAD

**Issue by kiwifb** [Singular/Singular#788]
Reported for a Gentoo prefix system at https://github.com/cschwan/sage-on-gentoo/issues/436 

```
libtool: compile:  x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -I. -I./include -I./include -I. -I/storage/strogdon/gentoo-rap/var/tmp/portage/sci-mathematics/singular-4.0.3_p3/work/singular-4.0.3 -I/storage/strogdon/gentoo-rap/var/tmp/portage/sci-mathematics/singular-4.0.3_p3/work/singular-4.0.3 -DSING_NDEBUG -DOM_NDEBUG -march=native -O0 -pipe -g -ggdb -fPIC -fno-common -O3 -Wno-unused-function -Wno-trigraphs -Wno-unused-parameter -Wunknown-pragmas -Wno-unused-variable -fomit-frame-pointer -fwrapv -fvisibility=default -finline-functions -fno-exceptions -fno-rtti -fno-threadsafe-statics -fno-enforce-eh-specs -fconserve-space -funroll-loops -fno-delete-null-pointer-checks -c cfGcdAlgExt.cc  -fPIC -DPIC -o .libs/cfGcdAlgExt.o
cfGcdAlgExt.cc: In function ‘void tryExtgcd(const CanonicalForm&, const CanonicalForm&, const CanonicalForm&, CanonicalForm&, CanonicalForm&, CanonicalForm&, bool&)’:
cfGcdAlgExt.cc:955:173: error: ‘void tryExtgcd(const CanonicalForm&, const CanonicalForm&, const CanonicalForm&, CanonicalForm&, CanonicalForm&, CanonicalForm&, bool&)’ was declared ‘extern’ and later ‘static’ [-fpermissive]
 static void tryExtgcd( const CanonicalForm & F, const CanonicalForm & G, const CanonicalForm & M, CanonicalForm & result, CanonicalForm & s, CanonicalForm & t, bool & fail )
                                                                                                                                                                             ^
In file included from cfGcdAlgExt.cc:27:0:
cfGcdAlgExt.h:29:6: note: previous declaration of ‘void tryExtgcd(const CanonicalForm&, const CanonicalForm&, const CanonicalForm&, CanonicalForm&, CanonicalForm&, CanonicalForm&, bool&)’
 void tryExtgcd( const CanonicalForm & F, const CanonicalForm & G, const CanonicalForm& M, CanonicalForm & result, CanonicalForm & s, CanonicalForm & t, bool & fail );
```

It looks like the `static` keyword was added in https://github.com/Singular/Sources/commit/c7afbd34f1aae46fbf47925ff2246e5f42f4ca08 but may be the corresponding header should have been changed as well.

**Comment by fingolfin** [comment]
I believe this issue can be closed?

**Comment by kiwifb** [comment]
Yes, it was fixed in 4.0.3_p4 somehow.
